public/app/plugins/datasource/grafana-azure-monitor-datasource/components/QueryEditor/QueryEditor.test.tsx

```diff
@@ -90,4 +90,22 @@ describe('Azure Monitor QueryEditor', () => {
     // reset config to not impact future tests
     config.featureToggles.azureMonitorResourcePickerForMetrics = originalConfigValue;
   });
+
+  it('should render the experimental QueryHeader when feature toggle is enabled', async () => {
+    const originalConfigValue = config.featureToggles.azureMonitorExperimentalUI;
+
+    config.featureToggles.azureMonitorExperimentalUI = true;
+
+    const mockDatasource = createMockDatasource();
+    const mockQuery = {
+      ...createMockQuery(),
+      queryType: AzureQueryType.AzureMonitor,
+    };
+
+    render(<QueryEditor query={mockQuery} datasource={mockDatasource} onChange={() => {}} onRunQuery={() => {}} />);
+
+    await waitFor(() => expect(screen.getByTestId('azure-monitor-experimental-header')).toBeInTheDocument());
+
+    config.featureToggles.azureMonitorExperimentalUI = originalConfigValue;
+  });
```

Comment:
Would be nice with a unit test that asserts that the as-is QueryTypeField is rendred in case the feature toggle is disabled/not set.

Reply:
Whoops, Github didn't refresh all the way so I missed this before merging. I'll include it in the next PR.